### PR TITLE
New tests

### PR DESCRIPTION
**Pull request by Ysabz**
### Summary 
this PR adds new units tests 
### Details
unit tests include
_ UI components
- jsonparsing
-cloud operations
# How to test?
_-test the app and run yarn test-coverage. (note: one test in cloud operation is currently failing which is due to some internal error)
### Checks 
- [x] Tested Changes
- [ ] Stakeholder Approval

### CODE REVIEW

File: src/test/cloudOperations.test.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import { getBlobsInContainer } from "../util/cloudOperations/readFromCloud";
+import saveExperienceToCloud from "../util/cloudOperations/writeToCloud";
+
+import {
+  getApparatusFromCloud,
+  getExperienceFromCloud,
+} from "../util/cloudOperations/readFromCloud";
```
Comment (mckenzietsean) [inline src/test/cloudOperations.test.tsx:7]: ReadFromCloud is getting imported twice.

File: src/test/cloudOperations.test.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  ) as jest.Mock;
+  const apparatusId = "wobble-sphere";
+  return getApparatusFromCloud(apparatusId)
+    .then((testoutput) => {
+      expect(testoutput).toEqual(metadata);
+    })
+});
+
+
+test("based on the user id and exp id, it should return the experience json but can reject", () => {
```
Comment (mckenzietsean) [inline src/test/cloudOperations.test.tsx:74]: Test fails on my end.

File: src/test/cloudOperations.test.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+test("based on the url, it should return the list of blobs", () => {
+  const id = "testuser1";
+  const output = ["testexp1"];
+  return getBlobsInContainer(id).then((testoutput) => {
+    expect(testoutput).toEqual(output);
+  });
+});
+
+test("save an experience to the cloud", () => {
```
Comment (mckenzietsean) [inline src/test/cloudOperations.test.tsx:118]: Test fails on my end.

**Review by mckenzietsean — COMMENTED** [review]
Generally everything is good aside from a couple of code smells and 2 failed tests. 37/39 of tests passing is still a successful amount.

File: src/test/cloudOperations.test.tsx (hunk 2 above)
Comment (shushengfuqin) [reply src/test/cloudOperations.test.tsx:74]: same for me. based on the test reaction. testoutput is null.
null and undefined are different.

**Review by shushengfuqin — COMMENTED** [review]
lgtm other than the 2 test failling

File: src/test/cloudOperations.test.tsx (hunk 2 above)
Comment (Ysabz) [reply src/test/cloudOperations.test.tsx:67]: fixed now

File: src/test/cloudOperations.test.tsx (hunk 3 above)
Comment (Ysabz) [reply src/test/cloudOperations.test.tsx:107]: this is failing due to some error in file creation which only happens in the test code and not in real run. I'll look into it

**Pull request merged**